common/views/components/IIIFPresentationPreview/IIIFPresentationPreview.js

```diff
@@ -0,0 +1,387 @@
+              };
+            }
+          })
+          .filter(Boolean);
+        return {
+          label: structure.label,
+          images,
+        };
+      })
+    : [];
+}
+
+function orderedStructuredImages(
+  structuredImages: IIIFThumbnails[]
+): IIIFThumbnails[] {
+  const titlePage = structuredImages.find(
+    structuredImage => structuredImage.label === 'Title Page'
+  );
+  const frontCover = structuredImages.find(
+    structuredImage =>
+      structuredImage.label === 'Front Cover' ||
+      structuredImage.label === 'Cover'
+  );
+  const tableOfContents = structuredImages.find(
+    structuredImage => structuredImage.label === 'Table of Contents'
+  );
+
+  const [firstImage] = tableOfContents ? tableOfContents.images : [];
+
+  return [
+    titlePage,
+    frontCover,
+    tableOfContents && { ...tableOfContents, images: [firstImage] },
+  ].filter(Boolean);
+}
+
+function previewThumbnails(
+  iiifManifest: IIIFManifest,
+  structuredImages: IIIFThumbnails[],
+  idealNumber: number = 5
+): IIIFThumbnails[] {
+  return structuredImages.length < idealNumber
+    ? structuredImages
+        .concat(
+          randomImages(
+            iiifManifest,
+            structuredImages,
+            idealNumber - structuredImages.length
+          )
+        )
+        .filter(x => {
+          return x.images.filter(image => image.id).length > 0;
+        })
+    : structuredImages.slice(0, idealNumber).filter(x => {
+        return x.images.filter(image => image.id).length > 0;
+      });
+}
+
+type Props = {|
+  itemUrl: any,
+  childManifestsCount?: number,
+|};
+
+type ViewType =
+  | 'unknown'
+  | 'multi'
+  | 'iiif'
+  | 'pdf'
+  | 'video'
+  | 'audio'
+  | 'none';
+// Can we show the user the work described by the manifest?
+// unknown === can't/haven't checked
+// iiif | pdf | video | audio === checked manifest and can render
+// none === checked manifest and know we can't render it
+type MultiVolumePreviewProps = {|
+  children: Node,
+|};
+
+const MultiVolumePreview = ({ children }: MultiVolumePreviewProps) => {
+  const multiPreview = useRef();
+  const isOnScreen = useOnScreen({ ref: multiPreview, threshold: 0.75 });
+
+  return (
+    <MultiVolumeContainer ref={multiPreview} isOnScreen={isOnScreen}>
+      {children}
+    </MultiVolumeContainer>
+  );
+};
+
+const IIIFPresentationPreview = ({
+  itemUrl,
+  childManifestsCount = 0,
+}: Props) => {
+  const [viewType, setViewType] = useState<ViewType>('unknown');
+  const [imageThumbnails, setImageThumbnails] = useState([]);
+  const [hasThumbnails, setHasThumbnails] = useState(false);
+  const [imageTotal, setImageTotal] = useState(0);
+  const iiifPresentationManifest = useContext(ManifestContext);
+
+  useEffect(() => {
+    const allImages = imageThumbnails.reduce(
+      (acc, thumb) => [...acc, ...thumb.images],
+      []
+    );
+    setHasThumbnails(allImages.length > 0);
+  }, [imageThumbnails]);
+
+  useEffect(() => {
```

Comment:
Do we need the `useEffect` here?

I thought `iiifPresentationManifest` should update outside of a `useEffect` as it's using the `useContext` book?